python/MooseDocs/extensions/content.py

```diff
@@ -281,3 +335,112 @@ def createHTML(self, parent, token, page):
+            nodes = [self.translator.findPage(p) for p in token['pages']]
+        else:
+            msg = "The 'location' and 'pages' tokens must be used exclusively."
+            raise exceptions.MooseDocsException(msg)
+
+        # Define convenience variables
+        hide = token['hide']
+        max_level = token['max_level']
+        if max_level > 6 or max_level < 1:
+            raise exceptions.MooseDocsException("The 'max_level' must be set in range of 1 to 6.")
+
+        # Create the outline from the headings of each node.
+        # This initializes the html tags to contain this list, where the previous heading level
+        # starts at zero thus will always trigger the creation of <ol><li>...</li></ol> within
+        # the "li" tag created here.
+        previous = 0 # initialize the previous heading level to zero to trigger creation of <ol><li> initially
+        li = html.Tag(parent, 'div', class_='moose-outline')
+        for node in nodes:
+            pageref = str(node.relativeDestination(page))
+
+            for key, head in node['headings'].items():
+                current = head['level']
+                if current <= max_level and key not in hide:
+                    diff = current - previous # change in heading number
+
+                    # At the same level, so add another <li> to the list
+                    if diff == 0:
+                        li = html.Tag(li.parent, 'li')
+
+                    # Heading level is less, so this heading goes within the current li
+                    elif diff > 0:
+                        for j in range(diff):
+                            li = html.Tag(html.Tag(li, 'ol'), 'li')
+
+                    # Heading level is more, so this heading goes above the current li
+                    else:
+                        ol = li.parent
+                        for i in range(-diff):
+                            ol = ol.parent.parent
+                        li = html.Tag(ol, 'li')
+
+                    url = pageref + '#{}'.format(key)
+                    link = core.Link(None, url=url)
+                    head.copyToToken(link)
+                    self.renderer.render(li, link, page)
+                    previous = current
+
+        # TODO: Make nested lists collapsible at their parent level
+        # The folowing reproduced an example from https://materializecss.com/collapsible.html
+        #
+        # ul = html.Tag(parent, 'ul', class_='collapsible')
+        # li = html.Tag(ul, 'li')
+        # div = html.Tag(li, 'div', class_='collapsible-header')
+        # html.Tag(div, 'i', class_='material-icons', string='filter_drama')
+        # html.Tag(div, string=' First')
+        # div = html.Tag(li, 'div', class_='collapsible-body')
+        # span = html.Tag(div, 'span', string='Lorem ipsum dolor sit amet.')
+
+    def createLatex(self, parent, token, page):
+        # TODO: Write Latex renderer - this capability will be useful for PDFs with several pages
```

Comment:
This should output a message (in Latex) that states that the command is not supported.

Reply:
Could I just throw a `MooseDocsException()`? Or do you mean in the latex document itself it should just create this message instead of an outline?

Reply:
The latex document should contain the error, for example https://github.com/idaholab/moose/blob/01ad7fec9f41f8f2e4329ebbfc62648d4a474d73/python/MooseDocs/extensions/media.py#L199